WordPress/pattern-directory
# Theme: Add favorites page

Fixes #239 - Add a page, `/favorites/`, which will show the current user's favorite patterns. This also updates PatternGridMenu a bit to be used on the Favorites page - now we can pass the current query down from Patterns, along with the basePath used for category links.

### Screenshots

All favorites, ex `wordpress.org/patterns/favorites`
<img width="1001" alt="all-fav" src="https://user-images.githubusercontent.com/541093/125674457-93dbab53-ec1b-4c32-b585-fc43d3308566.png">

Favorites in the Button category, ex `wordpress.org/patterns/favorites/pattern-categories/buttons`
<img width="1012" alt="buttons-fav" src="https://user-images.githubusercontent.com/541093/125674463-62d9cb92-8e5e-4860-9cc1-7d82a77d866d.png">

Favorites are localized, so only English patterns appear on the main site, while Japanese patterns appear on `ja.wordpress.org/patterns/favorites` (mirrors how My Patterns will work, can be iterated on post-launch if we decide this is not how it should work)
<img width="994" alt="ja-fav" src="https://user-images.githubusercontent.com/541093/125674465-bea25379-6a6a-4445-aabb-dba3800244d7.png">

### How to test the changes in this Pull Request:

1. Make sure you have a page with the slug `favorites`
2. Refresh permalinks, build the branch `yarn workspaces run build`
3. Favorite some patterns
3. View the favorites page, `/favorites/`
4. It should show all your favorites
5. Filter by category
6. It should only show those favorites in that category
7. The counts on the context bar should be correct
8. Unfavorite a pattern, it should be removed from the view
